DOC: Fixed implicit imports for whatsnew (v >= version 20.0)

saurav2608 opened the pull request:
- [x] xref #24177 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry
This partially resolves #24177 .

saurav-chakravorty committed:
DOC: Fixed implicit imports for whatsnew file >= version 20.0

datapythonista commented on doc/source/whatsnew/v0.23.1.rst:
```diff
@@ -43,19 +38,20 @@ To summarize, here's the behavior in 0.22.0, 0.23.0, 0.23.1:
 
    # 0.22.0... Silently coerce the datetime.date
-   >>> Series(pd.date_range('2017', periods=2)) == datetime.date(2017, 1, 1)
+   >>> import datetime as datetime
```
```suggestion
   >>> import datetime
```

datapythonista reviewed (commented):
Thanks @saurav2608 

Can we remove these files from the exclude in `setup.cfg`?

Do you mean in the PR title `>=0.23.0`

datapythonista committed:
Update doc/source/whatsnew/v0.23.1.rst

DOC: import datetime

Co-Authored-By: saurav2608 <sauravchakravorty@gmail.com>

saurav2608 commented:
> Can we remove these files from the exclude in setup.cfg?
Not yet, These changes only fix the implicit import errors. 
> Do you mean in the PR title >=0.23.0
I have fixed (if required) all the implicit import `from pandas import *` for version >= 20.0. I have removed the import block for all these files. flake8 errors still exist and I have opened individual issues for each of them.

datapythonista commented:
In this PR I only see files `>=0.23` changed, that's why I find the title misleading. Do you mean that in `0.20` to `0.22` we don't have the `import *`.

Do you think it's worth to fix the `import *` separate from the rest? Happy to merge this PR, but for the rest, for me it makes things simpler to fix all the problems in a single file at once, remove it from the exclusion list, and besides in the issue, keep track in `setup.cfg` of what is correct, and what needs to be fixed.

saurav2608 commented:
> In this PR I only see files `>=0.23` changed, that's why I find the title misleading. Do you mean that in `0.20` to `0.22` we don't have the `import *`.

Yes, that is correct. We don't have import *
> Do you think it's worth to fix the `import *` separate from the rest? Happy to merge this PR, but for the rest, for me it makes things simpler to fix all the problems in a single file at once, remove it from the exclusion list, and besides in the issue, keep track in `setup.cfg` of what is correct, and what needs to be fixed.

Okay. But please merge this for now.  For the rest I have not created issues yet. So, I can create an overall issues for all sorts of flake8 issues.

jreback commented:
@datapythonista seeing gazillion warnings in the doc build: https://travis-ci.org/pandas-dev/pandas/jobs/465964000 I don't think this PR is actually causing them, but should fix this up before any more doc fixes.

datapythonista commented:
Thanks for the heads up. Seems unrelated to the fixes we've been merging, but I'm taking a look now, I'll try to see what's wrong and send a fix asap.

TomAugspurger commented:
> @jreback the warnings come from #23601, where IntervalArrayFormatter was replaced by ExtensionArrayFormatter. But IntervalIndex is still using IntervalArrayFormatter.


That's fixed https://github.com/pandas-dev/pandas/pull/24134

saurav2608 commented:
@datapythonista  - are you waiting for #24134 to be merged before merging this?

datapythonista commented:
no, we like to double check what we merge, so I approved the changes, and whenever another maintainer have time to review this PR, they'll merge it if they're happy

jreback commented:
thanks @saurav2608 FYI don't put 'closes' in the PR header in cases like this, use xref (as github will close the referenced issue)
